build.gradle.kts

```diff
@@ -80,12 +80,19 @@ dependencies {
     api(libs.bundles.kotlin.qa)
     implementation(kotlin("stdlib-jdk8"))
     implementation(libs.kotlin.gradle.plugin.api)
-    testImplementation(gradleTestKit())
-    testImplementation(libs.konf.yaml)
-    testImplementation(libs.classgraph)
+    testImplementation(libs.testkit)
     testImplementation(libs.bundles.kotlin.testing)
 }
 
+inline fun <reified T : Task> Project.disableTrackState() {
+    tasks.withType<T>().configureEach {
+        doNotTrackState("Otherwise JaCoCo does not work correctly")
+    }
+}
+
+disableTrackState<Test>()
+disableTrackState<JacocoReport>()
```

Comment:
what are these for?

Reply:
Like you have to do [here](https://github.com/DanySK/Template-for-Gradle-Plugins/blob/f2993d87b62c3f93d1877e7c3dc4b36fe5ae4ad4/build.gradle.kts#L78), otherwise jacoco agent cannot access an own property and failed the tests.

If you're minding why not only on Windows, that's because in my CI disabling tracking state only on Windows, failed anyway and so I do not track them in any case.
We can try disabling them only on Windows, but I think tests would fail.

Attaching reminder failure below.
![issue](https://github.com/DanySK/gradle-kotlin-qa/assets/85103262/429c1e85-4887-40a6-afca-405337757c6f)

Reply:
Failed anyway... why? The workaround must be documented and understood. Can you provide a build scan of the failure under linux/osx?

Reply:
What failed?


Failed build Scan for Linux: https://scans.gradle.com/s/g4suhcesonofa
Failed build Scan for MacOs: https://scans.gradle.com/s/bytfnv5mtlvqc
Correct build for Windows: https://github.com/mirko-felice/gradle-plugin-testkit/actions/runs/5143820435/jobs/9259314630

All of these were built in the same commit, where I have disabled only the jacoco state on Windows. 
Because they threw the same error related to jacoco, as you can see in the scans, I thought disabling tracking state as you did would be a good workaround.

I was just miming your behavior in the same case.
What can I do to understand and document it?

Reply:
I just learnt that the issue is related to Gradle Daemon that is closed automatically by the Testkit official library, but jacoco agent does not wait for it.

It's an official issue of gradle/gradle#16603

Reply:
Great, then a comment in the build file explaining what's going on and a reference to the issue you mentioned would be fine!
The idea is that those who should read the build file in the future can understand why those lines are there.

Reply:
Thank you, done!